### DISCUSSION TITLE
In a flow how to send a mail to the user_created when an item is updated?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#16841] [Archive]
I get a `"Sendmail exited with code 75"`

**Comment by Pachat** [comment]
This [video](https://www.youtube.com/watch?v=8le81_SF4h4&t=395s) helped

I found out also how to compose the filter 
```
{
	"$trigger": {
		"payload": {
			"status": {
				"_eq": "answered"
			}
		}
	}
}

```

But still have 2 questions as on `item.update` the payload only triggers the modified values as we can see writing `{{$trigger}}` in the mail, which gives:
`"{"event":"question.items.update","payload":{"status":"answered"},"keys":["edbb61d0-0b42-4bac-a202-530b6a31c79c"],"collection":"question"}"`

What do I have to write

1. to include the `question.title` in the subject or in the body of the mail?
2. to set the email address to the `question.user_created.email`?

**Reply by paescuj** [reply]
You can configure your flow to something like this:
<img width="1063" alt="Screenshot 2022-12-17 at 21 47 58" src="https://user-images.githubusercontent.com/5363448/208265493-658008cb-337a-4947-abd7-b24bf824a8a2.png">
Note that you need to set the keys in the "Read Data" operations to the corresponding values (`question`, `user`).

**Reply by Pachat** [reply]
Thank you very much for your clear explanation.

As you said, I have to set the keys in the "Read Data" operations to the corresponding values (question, user).
In which "Read Data"? Is it in the Query? I am new to this and do not know how to do it. Any help would be appreciated.

**Reply by paescuj** [reply]
You're welcome!

There is a `Key` field in the options of the "Read Data" operation:
<img width="600" src="https://user-images.githubusercontent.com/5363448/208456356-d40f5ed7-2932-4bbd-ab6e-8791ff916566.png">

With the example flow from above, you need to set the key in the first "Read Data" operation to `question` and in the second one to `user`. This way the data is then available in subsequent steps under `{{question.title}}`, `{{user.email}}`, ...

**Reply by Pachat** [reply]
That simple! I thought, as a random name is created, that it was just some internal name.

Thank you very much for your answer which greatly helped.

**Reply by Pachat** [reply]
One last question:
I'd like to send a copy to the managers:
- How to send the email also to the users which role's name contains `Manager`?

The second `Read Data` on `Directus_users` has a field `IDs`
- How to fill it with the managers' emails?

I tried to leave it empty or set it to {{user.email}} with the query to
```
{
	"role": {
		"name": {
			"_contains": "Manager"
		}
	}
}
```
In both case I get the error
```
{
  "errors": [
    {
      "message": "Field \"resolve\" has to be unique.",
      "extensions": {
        "code": "RECORD_NOT_UNIQUE",
        "collection": "directus_operations",
        "field": "resolve",
        "invalid": "82455d61-1bb2-4f22-b2e5-0e686172d4cb"
      }
    }
  ]
}
```

**Reply by paescuj** [reply]
A bit late answer, but maybe still helpful for you or others:

Please note that you need to embed your filter configuration inside the `"filter"` property! In your case it could look like this to get the user from `user_created` along with all users with the `Manager` role:

```json
{
    "filter": {
        "_or": [
            {
                "id": {
                    "_eq": "{{ $last.user_created }}"
                }
            },
            {
                "role": {
                    "name": {
                        "_contains": "Manager"
                    }
                }
            }
        ]
    }
}
```

In a subsequent ["Run Script" operation](https://docs.directus.io/configuration/flows/operations.html#run-script) you could then prepare this data for the "Send Mail" operations:
```js
module.exports = function(data) {
  return data.$last.map((user) => user.email);
}
```

<img width="800" src="https://user-images.githubusercontent.com/5363448/216310894-862d951c-ca9e-4ad8-bdec-921a6baeac0f.png">
